### PR TITLE
Cleanups in "State recalculations"

### PR DESCRIPTION
**Pull request by JustinDrake**

### CODE REVIEW

File: specs/beacon-chain.md [hunk 1]
```diff
@@ -510,32 +510,34 @@ Repeat while `slot - last_state_recalculation_slot >= CYCLE_LENGTH`:
...
 #### Adjust justified slots and crosslink status
 
-For all slots `s` in `last_state_recalculation_slot - CYCLE_LENGTH ... last_state_recalculation_slot - 1`:
+For every slot `s` in the range `last_state_recalculation_slot - CYCLE_LENGTH ... last_state_recalculation_slot - 1`:
 
-* Determine the total set of validators that attested to that block at least once
-* Determine the total balance of these validators. If this value times three equals or exceeds the total balance of all active validators times two, set `last_justified_slot = max(last_justified_slot, s)` and `justified_streak += 1`. Otherwise, set `justified_streak = 0`
-* If `justified_streak >= CYCLE_LENGTH + 1`, set `last_finalized_slot = max(last_finalized_slot, s - CYCLE_LENGTH - 1)`
+* Determine the total balance of validators that attested to the beacon chain block at slot `s`.
+* If the above total balance times three equals or exceeds the total balance of all active validators times two, set `last_justified_slot = max(last_justified_slot, s)` and `justified_streak += 1`. Otherwise, set `justified_streak = 0`.
```
Comment (djrtwo) [inline specs/beacon-chain.md]: How about:

* Let `total_s_validator_balance` equal the total balance of validators that attested to the beacon chain block at slot `s`.
* If `3 * total_s_validator_balance >= 2* total_deposits`, set....

We could probably use a `total_deposits` helper too

File: specs/beacon-chain.md [hunk 2]
```diff
@@ -510,32 +510,34 @@ Repeat while `slot - last_state_recalculation_slot >= CYCLE_LENGTH`:
...
+For all `(shard, shard_block_hash)` tuples compute the total balance of validators that attested to the shard block with hash `shard_block_hash`. If this value times three equals or exceeds the total balance of all validators in the committee times two, and the current dynasty exceeds `crosslinks[shard].dynasty`, set `crosslinks[shard] = CrosslinkRecord(dynasty=dynasty, slot=block.last_state_recalculation_slot + CYCLE_LENGTH, hash=shard_block_hash)`.
 
 #### Balance recalculations related to FFG rewards
 
-Let `time_since_finality = block.slot - last_finalized_slot`, and let `B` be the balance of any given validator whose balance we are adjusting, not including any balance changes from this round of state recalculation. Let:
+* Let `total_deposits = sum([v.balance for i, v in enumerate(validators) if i in get_active_validator_indices(validators, dynasty)])`.
+* Let `total_deposits_in_eth = total_deposits // 10**9`.
```
Comment (djrtwo) [inline specs/beacon-chain.md]: probably define `10**9` as `GWEI_PER_ETH` in constants

Comment (JustinDrake) [reply specs/beacon-chain.md]: Yes, sounds good

File: specs/beacon-chain.md (hunk 1 above)
Comment (JustinDrake) [reply specs/beacon-chain.md]: Yes, will do

File: specs/beacon-chain.md (hunk 2 above)
Comment (JustinDrake) [reply specs/beacon-chain.md]: Done in this pull request https://github.com/ethereum/eth2.0-specs/pull/48

File: specs/beacon-chain.md [hunk 3]
```diff
@@ -510,45 +510,52 @@ Repeat while `slot - last_state_recalculation_slot >= CYCLE_LENGTH`:
...
+* If `3 * total_balance_attesting_at_s >= 2 * total_balance` set `last_justified_slot = max(last_justified_slot, s)` and `justified_streak += 1`. Otherwise set `justified_streak = 0`.
+* If `justified_streak >= CYCLE_LENGTH + 1` set `last_finalized_slot = max(last_finalized_slot, s - CYCLE_LENGTH - 1)`.
 
-For all (`shard`, `shard_block_hash`) tuples, compute the total deposit size of validators that attested to that block hash for that shard. If this value times three equals or exceeds the total balance of all validators in the committee times two, and the current dynasty exceeds `crosslinks[shard].dynasty`, set `crosslinks[shard] = CrosslinkRecord(dynasty=dynasty, slot=block.last_state_recalculation_slot + CYCLE_LENGTH, hash=shard_block_hash)`.
+For every `(shard, shard_block_hash)` tuple:
 
-#### Balance recalculations related to FFG rewards
+* Let `total_balance_attesting_to_h` be the total balance of validators that attested to the shard block with hash `shard_block_hash`.
+* Let `total_balance_attestable_to_h` be the total balance in the committee of validators that could have attested to the shard block with hash `shard_block_hash`.
```
Comment (djrtwo) [inline specs/beacon-chain.md]: I'd prefer `total_committee_balance`. Better readability

**Comment by djrtwo** [comment]
Yay progress! I'm going to handle the merge

**Pull request merged**